lib/_stream_wrap.js

```diff
@@ -49,7 +50,8 @@ function StreamWrap(stream) {
       this.pause();
       this.removeListener('data', ondata);
 
-      self.emit('error', new Error('Stream has StringDecoder'));
+      self.emit('error', new errors.Error('ERR_ASSERTION',
```

Comment:
@jasnell , I followed your suggestions and made those changes. 
In this process, I made few mistakes  but I guess I recovered everything, please have a look.
Thanks.